### DISCUSSION TITLE
C API/Interface

### DISCUSSION THREAD

**Discussion by general-rishkin** [google-research/dex-lang#627] [Q&A] [answered]
I see that Dex can now be called from Julia. Is there a way to expose a C API of some sort or compile to C like Futhark so that Dex can be used with FFI in other languages?

**Comment by oxinabox** [comment, marked as answer]
(Not yet from Julia, PR is still open)

There is a C API.
That is what the Julia PR is using and what Python is using.

**Reply by general-rishkin** [reply]
Excellent. Could you possibly point me to it. I cannot find it.

**Reply by oxinabox** [reply]
https://github.com/google-research/dex-lang/blob/main/src/Dex/Foreign/API.hs

**Reply by general-rishkin** [reply]
Thanks.

**Reply by apaszke** [reply]
Also if you find yourself wishing for more functions, let us know. The C API is very preliminary and also not very stable, but should be good enough for prototype integrations.
